util/predicates/generic_predicates.go

```diff
@@ -167,3 +193,19 @@ func processIfNotPaused(logger logr.Logger, obj client.Object) bool {
 	log.V(4).Info("Resource is not paused, will attempt to map resource")
 	return true
 }
+
+func processIfLabelMatch(logger logr.Logger, obj client.Object, labelValue string) bool {
```

Comment:
Maybe this name should indicate that it's only looking for the watch label or else allow the `HasWatchLabel` function to be passed in as an argument.

Reply:
Do you have any suggestion for naming? I think adding the function as an argument is going to be unnecessarily complex.